lib/hls/hls_parser.js

```diff
@@ -928,13 +928,12 @@ shaka.hls.HlsParser = class {
     if (!playlist.segments.length) {
       return defaultBasicInfo;
     }
-    const middleSegmentIdx = Math.trunc((playlist.segments.length - 1) / 2);
-    const middleSegment = playlist.segments[middleSegmentIdx];
-    const middleSegmentUris = shaka.hls.Utils.constructSegmentUris(
+    const firstSegment = playlist.segments[0];
```

Comment:
The change in which we started looking for a segment in the middle of the playlist, instead of the first one, was originally written by you.
In all honesty I don't remember why we were doing it before. I suppose I'll have to take your word that it is no longer necessary.

Reply:
It's not really necessary anymore and it also causes problems with byterange (we would have to look at all the previous segments to calculate the range).